File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/GoodsDescService.java
package com.pinyougou.sellergoods.service;
import java.util.List;
import com.pinyougou.pojo.TbGoodsDesc;

import entity.PageResult;
/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface GoodsDescService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbGoodsDesc> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(TbGoodsDesc goodsDesc);
	
	
	/**
	 * 修改
	 */
	public void update(TbGoodsDesc goodsDesc);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbGoodsDesc findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long [] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TbGoodsDesc goodsDesc, int pageNum,int pageSize);
	
}

File: pinyougou-manager-web/src/main/webapp/js/service/goodsDescService.js
//服务层
app.service('goodsDescService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../goodsDesc/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../goodsDesc/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../goodsDesc/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../goodsDesc/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../goodsDesc/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../goodsDesc/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../goodsDesc/search.do?page='+page+"&rows="+rows, searchEntity);
	}    	
});
